cmd/utils/cmd.go

```diff
@@ -63,7 +65,21 @@ func Fatalf(format string, args ...interface{}) {
 	os.Exit(1)
 }
 
+func IsPermissionEnabled(ctx *cli.Context) bool {
+	if ctx.GlobalBool(EnableNodePermissionFlag.Name) {
+		fileName := "permission-config.json"
+		fullPath := filepath.Join(ctx.GlobalString(DataDirFlag.Name), fileName)
```

Comment:
CLI flag `--datadir` is indeed optional and has a default value. Further, `datadir` can be also provided via TOML config file using `--config` flag. 

Please pass `*node.Config` to this method and obtain `datadir` from it.

Reply:
`IsPermissionEnabled` is called from two places - `cmd.go` and `main.go`. Hence changed the function signature to pass the two values it is looking for, datadir and permissionEnabled flag.